test/e2e/categories_test.go

```diff
@@ -0,0 +1,186 @@
+		Expect(namespace).NotTo(BeNil())
+		flavor := clusterctl.DefaultFlavor
+		By("Creating a workload cluster (no additional categories)")
+		deployClusterAndWait(
+			deployClusterParams{
+				clusterName:           clusterName,
+				namespace:             namespace,
+				flavor:                flavor,
+				clusterctlConfigPath:  clusterctlConfigPath,
+				artifactFolder:        artifactFolder,
+				bootstrapClusterProxy: bootstrapClusterProxy,
+				e2eConfig:             *e2eConfig,
+			}, clusterResources)
+
+		By("Checking cluster category condition is true")
+		verifyConditionOnNutanixCluster(verifyConditionOnNutanixClusterParams{
+			clusterName:           clusterName,
+			namespace:             namespace,
+			bootstrapClusterProxy: bootstrapClusterProxy,
+			expectedCondition: clusterv1.Condition{
+				Type:   infrav1.ClusterCategoryCreatedCondition,
+				Status: corev1.ConditionTrue,
+			},
+		})
+
+		By("Checking if a category was created")
+		expectedCategoryKey := getExpectedClusterCategoryKey(clusterName)
+		_, err := nutanixClient.V3.GetCategoryValue(expectedCategoryKey, defaultClusterCategoryValue)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("Checking if there are VMs assigned to this category")
+		expectedCategories := map[string]string{
+			expectedCategoryKey: defaultClusterCategoryValue,
+		}
+		verifyCategoriesNutanixMachines(clusterName, namespace.Name, expectedCategories)
+
+		By("PASSED!")
+	})
+	It("Create a cluster with additional categories", func() {
+		Expect(namespace).NotTo(BeNil())
+		flavor := "additional-categories"
+
+		By("Creating a workload cluster with additional categories set")
+		deployClusterAndWait(
+			deployClusterParams{
+				clusterName:           clusterName,
+				namespace:             namespace,
+				flavor:                flavor,
+				clusterctlConfigPath:  clusterctlConfigPath,
+				artifactFolder:        artifactFolder,
+				bootstrapClusterProxy: bootstrapClusterProxy,
+				e2eConfig:             *e2eConfig,
+			}, clusterResources)
+
+		By("Assign additional categories to the vms")
+		expectedCategoryKey := getExpectedClusterCategoryKey(clusterName)
+		additionalCategory := getAdditionalCategoryFromEnvironment()
+		Expect(additionalCategory.key).ToNot(BeEmpty())
+		Expect(additionalCategory.value).ToNot(BeEmpty())
+		expectedCategories := map[string]string{
+			expectedCategoryKey:    defaultClusterCategoryValue,
+			additionalCategory.key: additionalCategory.value,
+		}
+		verifyCategoriesNutanixMachines(clusterName, namespace.Name, expectedCategories)
+
+		By("PASSED!")
+	})
+
+	It("Create a cluster linked to non-existing categories (should fail)", func() {
+		flavor = "no-nmt"
+		Expect(namespace).NotTo(BeNil())
+
+		By("Creating Nutanix Machine Template with invalid categories")
+		invalidProjectNMT := createDefaultNMT(clusterName, namespace.Name)
+		invalidProjectNMT.Spec.Template.Spec.AdditionalCategories = []infrav1.NutanixCategoryIdentifier{
+			infrav1.NutanixCategoryIdentifier{
```

Comment:
can we also add a test with multiple items in the list of categories instead of just one.

Reply:
Modified